app/jobs/petition_pdf_generation_worker.rb

```diff
@@ -16,7 +16,9 @@ def perform(sqs_msg, body)
 
     version = repository.find_by_id! petition_detail_version_id
 
-    document_url = petition_pdf_service.generate(version) 
-    version.update published: true, document_url: document_url
+    generated_version = petition_pdf_service.generate(version) 
+    version.update published: true, document_url: generated_version.document_url, sha: generated_version.sha
```

Comment:
We should only publish the version after the document has been certified on the blockchain.

The mobile api must push a message into a queue which we will be listening so we can publish the version.